PollyDemos/Async/AsyncDemo07_WaitAndRetryNestingCircuitBreakerUsingPipeline.cs

```diff
@@ -0,0 +1,159 @@
+                    return default;
+                },
+                OnClosed = args =>
+                {
+                    progress.Report(ProgressWithMessage(".Breaker logging: Call OK! Closed the circuit again!", Color.Magenta));
+                    return default;
+                },
+                OnHalfOpened = args =>
+                {
+                    progress.Report(ProgressWithMessage(".Breaker logging: Half-open: Next call is a trial!", Color.Magenta));
+                    return default;
+                }
+            }); // New for demo07: here we are not calling the Build method because we want to add one more strategy to the pipeline.
+
+            // Define our retry strategy:
+            pipelineBuilder.AddRetry(new()
+            {
+                // Exception filtering - we don't retry if the inner circuit-breaker judges the underlying system is out of commission.
+                ShouldHandle = new PredicateBuilder().Handle<Exception>(ex => ex is not BrokenCircuitException),
+                MaxRetryAttempts = int.MaxValue, // Retry indefinitely
+                Delay = TimeSpan.FromMilliseconds(200),  // Wait between each try
+                OnRetry = args =>
+                {
+                    // Due to how we have defined ShouldHandle, this delegate is called only if an exception occurred.
+                    // Note the ! sign (null-forgiving operator) at the end of the command.
+                    var exception = args.Outcome.Exception!; // The Exception property is nullable
+
+                    // Tell the user what happened
+                    progress.Report(ProgressWithMessage($"Strategy logging: {exception.Message}", Color.Yellow));
+                    retries++;
+                    return default;
+                }
+            }); // New for demo07: here we are not calling the Build method because we will do it as a separate step to make the code cleaner.
+
+            // New for demo07: here we build the pipeline since we have added all the necessary strategies to it.
+            var pipeline = pipelineBuilder.Build();
+
+            var client = new HttpClient();
+            var internalCancel = false;
+
+            // Do the following until a key is pressed
+            while (!(internalCancel || cancellationToken.IsCancellationRequested))
+            {
+                totalRequests++;
+                var watch = new Stopwatch();
+                watch.Start();
+
+                try
+                {
+                    // Manage the call according to the pipeline.
+                    var responseBody = await pipeline.ExecuteAsync(async token =>
+                    {
+                        // This code is executed through both strategies in the pipeline:
+                        // Retry is the outer, and circuit breaker is the inner.
+                        // Demo 06 shows a broken-out version of what this is equivalent to.
```

Comment:
As at first reading I thought the "broken" meant "doesn't work".

```suggestion
                        // Demo 06 shows a decomposed version of what this is equivalent to.
```